src/api/Dockerfile

```diff
@@ -1,30 +1,75 @@
-# Node Alpine Docker Image
-FROM node:18.9-alpine3.16
+###################
+# 1. BUILD
+###################
 
-# Variables
+# NPM 8.19.1 Alpine linux image
+FROM node:18.15-alpine3.16 AS build
+
+# Build variables
 ARG GITHUB_SHA
 ARG SESSION_SECRET
 ARG GOV_NOTIFY_API_KEY
-ENV GITHUB_SHA $GITHUB_SHA
-ENV SESSION_SECRET $SESSION_SECRET
-ENV GOV_NOTIFY_API_KEY $GOV_NOTIFY_API_KEY
+ARG PORT
+ENV GITHUB_SHA=$GITHUB_SHA
+ENV SESSION_SECRET=$SESSION_SECRET
+ENV GOV_NOTIFY_API_KEY=$GOV_NOTIFY_API_KEY
+ENV PORT=$PORT
 
-# Install packages
-RUN apk add bash curl
-RUN rm -rf /var/cache/apk/*
 
-# Node setup
 WORKDIR /app
 
-COPY package.json .
-COPY package-lock.json .
-COPY .keystone keystone
-
-RUN npm i
+# NPM
+COPY --chown=node:node package.json .
+COPY --chown=node:node package-lock.json .
+RUN npm ci --legacy-peer-deps --ignore-scripts
 RUN npm cache clean --force
 
-COPY . .
+# Copy API
+COPY --chown=node:node . .
+
+# Build with all dependencies
 RUN npm run build
 
+# Lean NPM - Only install `dependencies`
+# `devDependencies` will still be resolved inside `package-lock.json`,
+# however they will not be installed inside `node_modules` directory.
+RUN npm ci --legacy-peer-deps --omit=dev --ignore-scripts
+
+# Non-root user
+USER node
+
+###################
+# 2. PRODUCTION
+###################
+
+# NPM 9.5.1 Alpine linux image
+FROM node:19.8-alpine3.16 AS production
```

Comment:
this version of node is different than the one from above. shouldn't both of them be the same?

Reply:
Build image has quite number of vulnerabilities, where as the `production` base image has only `1`.

Reply:
cool. but shouldn't we use the same version from prod to build? i.e. both versions should be 19.8?